xcarchive/ios.go

```diff
@@ -378,3 +379,38 @@ func (archive IosArchive) BundleIDProfileInfoMap() map[string]profileutil.Provis
 func (archive IosArchive) FindDSYMs() ([]string, []string, error) {
 	return findDSYMs(archive.Path)
 }
+
+// ReadCodesignParameters ...
+func (archive IosArchive) ReadCodesignParameters() (*autocodesign.AppLayout, error) {
+	var teamID string
+	entitlementsMap := map[string]autocodesign.Entitlements{}
+
+	bundleIDProfileInfoMap := archive.BundleIDProfileInfoMap()
+	for bundleIdentifier, provisioningProfile := range bundleIDProfileInfoMap {
+		if teamID == "" {
+			teamID = provisioningProfile.TeamID
+		}
+
+		entitlementsMap[bundleIdentifier] = autocodesign.Entitlements(provisioningProfile.Entitlements)
+	}
+
+	var platform autocodesign.Platform
+
+	platformName := archive.Application.InfoPlist["DTPlatformName"]
```

Comment:
Can maybe this and https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-application-loader/blob/245a973d4c4655ddf37ff491c3e52d17ebec6259/main.go#L76 be extracted to the same helper function?

Reply:
What do you think, what and how should we share here?

As I see, only this tiny part could be shared: https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-application-loader/blob/245a973d4c4655ddf37ff491c3e52d17ebec6259/main.go#L91-L107

But, in the case of an xcarchive `*simulator` platforms are not valid. Also `watchos` does not seem to be a valid platform if you read an archive file (even if you are building a standalone watchos application the platform name is: `iphoneos`)

Reply:
Thanks for checking, based on this you can leave as is.